include/private/UploadArtifactTask.h

```diff
@@ -14,15 +14,15 @@ namespace prism
 namespace connect
 {
 
-class ArtifactUploadHelper;
+class Client;
 
 class UploadArtifactTask
 {
 public:
     virtual ~UploadArtifactTask()
     {}
 
-    virtual Status execute(ArtifactUploadHelper* uploader) const = 0;
+    virtual Status execute(Client& client, id_t accountId, id_t cameraId) const = 0;
```

Comment:
After some thought, I think second option will result in the least code per method added.